### DISCUSSION TITLE
Messaging Request Feature

### DISCUSSION THREAD

**Discussion by cleandevcode** [GetStream/stream-chat-react-native#1479] [Q&A] [answered]
Hello.
I am using this SDK for my RN app and it's really amazing.
I am gonna implement one feature for messaging requests.
So the idea is that there is a request to another user and he can approve or deny my request.
When checking the document, I found the event custom - https://getstream.io/chat/docs/react-native/custom_events/

I am think if I can use this `friendship_request` feature 
![image](https://user-images.githubusercontent.com/52365167/173929375-0cf17084-a92a-4763-bbeb-c9abd22905fd.png)

But actually this type did not exist.
I could not find it from EVENT_MAP.
![image](https://user-images.githubusercontent.com/52365167/173929818-bcac0584-ca01-4555-93af-6058f1105dbf.png)

But I am not sure if I can use this feature for my idea.
If I can use it, then which event type is for it?

```
stream-chat: 6.1.0
stream-chat-expo: ^4.1.0
react: 16.15.1
expo: ^40.0.0
react-natve: 0.63.4
```

Thanks.

**Comment by khushal87** [comment, marked as answer]
Hey @cleandevcode, the purpose of custom events is to have more events other than the one mentioned on the map(they are the default ones that we provide to our customers). You can follow the following guides:

- Creating Custom events - https://getstream.io/chat/docs/react-native/custom_events/
- The event object where in you can add your custom data too - https://getstream.io/chat/docs/react-native/event_object/
- How to listen to custom events? - https://getstream.io/chat/docs/react-native/event_listening/

Hope this might help you.
